better docs for compiler - related #98 (#207)
grpc/grpc-kotlin: compiler/README.md

gRPC Kotlin Codegen Plugin for Protobuf Compiler
================================================

This generates the Kotlin interfaces out of the service definition from a
`.proto` file. It works with the Protobuf Compiler (`protoc`).

### Build Tool Plugins

> Usually this compiler is used via a built took plugin, like in Gradle, Maven, etc.

Reviewer: Should be `build tool plugin`?

Author: ```suggestion
Usually this compiler is used via a build tool plugin, like in Gradle, Maven, etc.
```

> For Gradle include the [protobuf plugin](https://github.com/google/protobuf-gradle-plugin) with at least version `0.8.13`, like:

Reviewer: nit: add comma: `For Gradle, include...`

Author: ```suggestion
For Gradle, include the [protobuf plugin](https://github.com/google/protobuf-gradle-plugin) with at least version `0.8.13`, like:
```

```
plugins {
    id("com.google.protobuf") version "SOME_VERSION"
}
```

Add dependencies on either `grpc-kotlin-stub` or `grpc-kotlin-stub-lite` like:
```
dependencies {
    implementation("io.grpc:grpc-kotlin-stub:SOME_VERSION")
}
```

Finally, setup the protobuf plugin:
```
protobuf {
    protoc {
        artifact = "com.google.protobuf:protoc:SOME_VERSION"
    }
    plugins {
        id("grpc") {
            artifact = "io.grpc:protoc-gen-grpc-java:SOME_VERSION"
        }
        id("grpckt") {
            artifact = "io.grpc:protoc-gen-grpc-kotlin:SOME_VERSION:jdk7@jar"
        }
    }
    generateProtoTasks {
        ofSourceSet("main").forEach {
            it.plugins {
                id("grpc")
                id("grpckt")
            }
        }
    }
}
```

> For Maven include the [protobuf plugin](https://www.xolstice.org/protobuf-maven-plugin/) and configure it to use the

Reviewer: Same here, `For Maven, include...`

Author: ```suggestion
For Maven, include the [protobuf plugin](https://www.xolstice.org/protobuf-maven-plugin/) and configure it to use the
```
